libazureinit/src/provision/user.rs

```diff
@@ -96,28 +100,50 @@ impl Provisioner {
     }
 }
 
+#[instrument(skip_all)]
+fn user_exists(username: &str) -> Result<bool, Error> {
+    let status = Command::new("getent")
+        .arg("passwd")
+        .arg(username)
+        .status()?;
+
+    Ok(status.success())
+}
+
 #[instrument(skip_all)]
 fn useradd(user: &User) -> Result<(), Error> {
-    let path_useradd = env!("PATH_USERADD");
-    let home_path = format!("/home/{}", user.name);
-
-    let status = Command::new(path_useradd)
-                    .arg(&user.name)
-                    .arg("--comment")
-                    .arg(
-                      "Provisioning agent created this user based on username provided in IMDS",
-                    )
-                    .arg("--groups")
-                    .arg(user.groups.join(","))
-                    .arg("-d")
-                    .arg(home_path)
-                    .arg("-m")
-                    .status()?;
-    if !status.success() {
-        return Err(Error::SubprocessFailed {
-            command: path_useradd.to_string(),
-            status,
-        });
+    const MAX_RETRIES: u8 = 5;
+
+    if user_exists(&user.name)? {
+        return Ok(());
+    }
+
+    for attempt in 1..=MAX_RETRIES {
```

Comment:
I'm not sure if it's worth retrying. I am not able to come up with a scenario where useradd would succeed after trying again while provisioning. If useradd fails, we should log an error and return an error at the end